test/API/H5_api_async_test.c

```diff
@@ -2662,71 +2669,65 @@ test_file_reopen(void)
     }
     H5E_END_TRY
 
-    return 1;
+    return;
 } /* end test_file_reopen() */
 
 /*
  * Cleanup temporary test files
  */
 static void
-cleanup_files(void)
+test_file_cleanup(const void H5_ATTR_UNUSED *params)
 {
-    char file_name[64];
-    int  i;
+    if (GetTestCleanup()) {
+        char file_name[64];
+        int  i;
 
-    remove_test_file(NULL, ASYNC_API_TEST_FILE);
+        remove_test_file(NULL, ASYNC_API_TEST_FILE);
```

Comment:
Creation of this file is skipped based on VOL cap flags, so cleanup probably should be too